ArcGISRuntimeSDKQt_CppSamples/Layers/ExportVectorTiles/ExportVectorTiles.qml

```diff
@@ -0,0 +1,163 @@
+                font.pixelSize: 16
+            }
+
+            Text {
+                id: progressText
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: model.exportProgress + "% Completed"
+                font.pixelSize: 16
+            }
+
+            onWidthChanged: {
+                exportProgressWindow.width = exportProgressColumn.width + 20
+            }
+
+            onHeightChanged: {
+                exportProgressWindow.height = exportProgressColumn.height + 20
+            }
+        }
+    }
+
+    // Create an extent rectangle for selecting the offline area
+    Rectangle {
+        id: extentRectangle
+        anchors.centerIn: parent
+
+        width: parent.width - parent.width * 0.1
+        height: parent.height - parent.height * 0.25
+        color: "transparent"
+        border {
+            color: "red"
+            width: 3
+        }
+
+        visible: model.jobStatus === 0 || model.jobStatus === 4
+    }
+
+    // Button to start the download
+    Button {
+        id: button
+        anchors {
+            bottom: parent.bottom
+            bottomMargin: item.height * .05
+            horizontalCenter: parent.horizontalCenter
+        }
+        width: 150
+
+        text: "Export area"
+
+        onClicked: {
+            switch(model.jobStatus) {
+            case 0: // Not started
+                model.startExport(extentRectangle.x, (extentRectangle.y + extentRectangle.height), (extentRectangle.x + extentRectangle.width), extentRectangle.y);
+                break;
+            case 1: // Started
+                model.cancel();
+                break;
+            case 2: // Paused
+                break;
+            case 3: // Succeeded
+                model.reset();
+                break;
+            case 4: // Failed
+                model.startExport(extentRectangle.x, (extentRectangle.y + extentRectangle.height), (extentRectangle.x + extentRectangle.width), extentRectangle.y);
+                break;
+            case 5: // Cancelling
+                break;
+            default:
+                break;
+            }
+        }
+    }
+
+    // Declare the C++ instance which creates the scene etc. and supply the view
+    ExportVectorTilesSample {
+        id: model
+        mapView: view
+
+        onJobStatusChanged: {
+            switch(jobStatus) {
+            case 0: // Not started
+                button.text = "Export area"
+                break;
+            case 1: // Started
+                button.text = "Cancel export"
+                break;
+            case 2: // Paused
+                break;
+            case 3: // Succeeded
+                button.text = "Reset"
+                break;
+            case 4: // Failed
+                button.text = "Export area"
+                break;
+            case 5: // Cancelling
+                break;
+            default:
+                break;
```

Comment:
Rather than adding a comment explaining each case, could you create a list of enums to reflect what each number indicates?

```C++
enum Status { NotStarted , Started, Paused, Succeeded, Failed, Cancelling };
```